src/OpenTelemetry.Exporter.OpenTelemetryProtocol/LogRecordExportProcessorOptions.cs

```diff
@@ -0,0 +1,79 @@
+// <copyright file="LogRecordExportProcessorOptions.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Diagnostics;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using OpenTelemetry.Internal;
+using OpenTelemetry.Logs;
+
+namespace OpenTelemetry.Exporter;
+
+/// <summary>
+/// Options for configuring either a <see cref="SimpleLogRecordExportProcessor"/> or <see cref="BatchLogRecordExportProcessor"/>.
+/// </summary>
+public class LogRecordExportProcessorOptions
```

Comment:
Shouldn't this class be in the SDK rather than the OTLP exporter project and under the `OpenTelemetry.Logs` namespace? This is akin to `MetricReaderOptions`.

Reply:
This options by itself cannot be used with any SDK component so not sure if it should be part of SDK as a public API.

Is their any other exporter that can use this? `MetricReaderOptions` is used by Console\Otlp.

Reply:
> This options by itself cannot be used with any SDK component so not sure if it should be part of SDK as a public API.

We have other examples of this. For example, `BatchExportActivityProcessorOptions` is in the SDK but not used directly by anything in the SDK project. We should stay consistent with our options classes and keep them close to the thing being configured - in this case a log processor.

> Is their any other exporter that can use this? MetricReaderOptions is used by Console\Otlp.

It's fine if the OTLP exporter is the only one that uses them. Consistency is what's important. I could write my own log exporter and benefit from the `LogRecordExportProcessorOptions`.

Reply:
> It's fine if the OTLP exporter is the only one that uses them. Consistency is what's important. I could write my own log exporter and benefit from the LogRecordExportProcessorOptions.

I think this whole thing started because we were constrained by the need to stay consistent with the existing bad patterns of OTLP Exporter. For anyone trying to write their custom exporter, they can always follow better options (For e.g. [OneCollector](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/src/OpenTelemetry.Exporter.OneCollector/Logs/OneCollectorLogExportProcessorBuilder.cs#L55-L63)) to offer configurability than to follow what OTLP Log Exporter is trying to do here. All that they really need is `BatchExportLogRecordProcessorOptions` which is already offered by the SDK.

I think we should keep this in OTLP package to:
- Avoid bloating the SDK public API surface with things that are not really needed by our users
- Discourage adoption of patterns followed in OTLP Exporter

Reply:
> I think this whole thing started because we were constrained by the need to stay consistent with the existing bad patterns of OTLP Exporter.

The issue of where this class belongs should not be conflated with our "bad patterns" discussion. This is about code organization not patterns that make it challenging for us to support certain features (e.g., DI and signal-specific environment variables).

Pre-dating this discussion. @CodeBlanch intuited the organization I am proposing here. He placed this class [in the SDK on the `main-logs` branch](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main-logs/src/OpenTelemetry/Logs/ExportLogRecordProcessorOptions.cs). Taking this a step further he also added a corresponding class for [traces on the `main-logs` branch](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main-logs/src/OpenTelemetry/Trace/ExportActivityProcessorOptions.cs).

Similarly, I proposed this same organization on the [logs "blueprint" PR I submitted](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4416).

The organization that @CodeBlanch and I have come to stems from how we have historically organized similar classes across all signals since our 1.0.0 release.

All of the following exist in the SDK project today:

* https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/BatchExportProcessorOptions.cs
* https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/Trace/BatchExportActivityProcessorOptions.cs
* https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/Metrics/MetricReaderOptions.cs
* https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/Metrics/PeriodicExportingMetricReaderOptions.cs
* https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/Logs/BatchExportLogRecordProcessorOptions.cs

Was it the wrong decision to put these classes in the SDK? Maybe, but I think it would be a far worse decision begin to spaghetti 🍝 how we organize our code.

Reply:
I think there is a major difference between the already existing SDK options and the new `LogRecordExportProcessorOptions`.The existing SDK options are targeted towards a specific individual SDK component and not some mix and match of these components. For example, `BatchExportActivityProcessorOptions` target only a single "component" which is `BatchActivityExportProcessor`.   We haven't exposed options that try to target a combination of SDK components like we are doing in this PR where we use a **single** options class that targets **both** `SimpleLogRecordExportProcessor` and `BatchLogRecordExportProcessor`.

Anyone who has a custom requirement of configuring multiple components together should be able combine the existing options themselves to achieve the same. Why should this particular combination of targeting `Simple` and `Batch` options together be given special consideration?

<br>


> The issue of where this class belongs should not be conflated with our "bad patterns" discussion. 

The need for this class arose only because of OTLP Exporter trying to maintain some consistency in the usage of its extension methods, right? If the class is a result of OTLP Exporter's constraints, then I think it should belong to OTLP Exporter project.

Reply:
My thoughts...

I did consider the current location before approving but I didn't bring it up because if we put `LogRecordExportProcessorOptions` into OtlpExporter now we can always move it to SDK later using [TypeForwardedToAttribute](https://learn.microsoft.com/dotnet/api/system.runtime.compilerservices.typeforwardedtoattribute).

In my original work I did put this into SDK. I also had a different name `ExportLogRecordProcessorOptions`. The name is screwy but it fit better with what had already been done I thought. The reason I had it in SDK was because I also had an [AddExporter](https://github.com/open-telemetry/opentelemetry-dotnet/blob/5be8226861c12b021bc1ea65516ea05755e8c1f9/src/OpenTelemetry/Logs/Builder/LoggerProviderBuilderExtensions.cs#L116) API using it. That was later removed.

Reply:
Sure we could use TypeForwardedTo in the future, but at a minimum the namespace `OpenTelemetry.Logs` should be considered. But to be honest, I can't get behind the placement of this class in the OTLP exporter project 😆.

Putting our differences aside for a moment, though, here's another idea: don't have this class at all.

I would be entirely ok with

```csharp
public static LoggerProviderBuilder AddOtlpExporter(
    this LoggerProviderBuilder builder,
    Action<OtlpExporterOptions, BatchExportLogRecordProcessorOptions> configureExporterAndProcessor)
```

I am also ok telling folks who _really_ want to use the SimpleExportProcessor combined with the OTLP exporter to do

```csharp
loggerProviderBuilder.AddProcessor(new SimpleLogRecordExportProcessor(new OtlpLogExporter()))
```

Reply:
Good point. Agree with using `OpenTelemetry.Logs` for it! Don't feel passionately about where it goes. Maybe leaning more towards putting it in SDK. Only because it seems useful for other exporters which may want to expose toggling simple or batch modes.

Reply:
How about changing the namespace to `OpenTelemetry.Logs` and keeping it in otlp. In future if needed we can use `TypeForwardedTo`? 

@alanwest For the second idea, that would mean we won't support options via DI for simple export processor (deviating from tracing experience).

Reply:
> How about changing the namespace to OpenTelemetry.Logs and keeping it in otlp. In future if needed we can use TypeForwardedTo?

Here are my current thoughts:
1. Move the class to `OpenTelemetry.Logs` namespace.
2. I do not approve of the location of this class because all other processor-related options classes reside in the SDK as well as the `ExportProcessorType` enum. However, you have two other maintainers approval, so let's merge to continue making progress towards an alpha release.
3. I will open an issue to continue this discussion prior to 1.6 release.

> For the second idea, that would mean we won't support options via DI for simple export processor (deviating from tracing experience).

I'm not sure what you mean. The plan we've discussed involves doing for traces what we end up doing for logs. That is, if we were to adopt the second idea, I expect there also to be an extension method added for traces

```csharp
public static TracerProviderBuilder AddOtlpExporter(
    this TracerProviderBuilder builder,
    Action<OtlpExporterOptions, BatchExportActivityProcessorOptions> configureExporterAndProcessor)
```

We have also discussed obsoleting the processor-related options on `OtlpExporterOptions`.